File: jdocs/Comp/Lectures/Lecture1/Homework/R/Homework1.R
# -----------------------------------------------------------------------------
# Exercise 1
# -----------------------------------------------------------------------------

# Simple solution
v1 = c(1,2,3,4,5,6,7,8,7,6,5,4,3,2,1)
# More elegant would be this here:
x=seq(1,8, by=1)
y=seq(7,1, by=-1)
v1=c(x,y)
cat('v1 = ', v1, "\n")

# Second vector simple
v2 = c(1,2,2,3,3,3,4,4,4,4,5,5,5,5,5)
# Or more elegant, how for such a short vector the extra work doesn't pay off
v2=c(rep(1,1),rep(2,2), rep(3,3), rep(4,4), rep(5,5))
cat('v2 = ', v2, "\n")

# Matrix
m1 = rbind(c(0,1,1),c(1,0,1),c(1,1,0))
m2 = rbind(c(0,2,3),c(0,5,0),c(7,0,0))
m1
m2

# -----------------------------------------------------------------------------
# Exercise 2
# -----------------------------------------------------------------------------

p=5.0  # 5% ! Do not use 0.05 here, otherwise the formula is incorrect.
A=1000.0
n=3
final_amount=A*(1.0+(p/100))^n 
cat("\n\n")
cat("The final amount is", final_amount, "euros")


# -----------------------------------------------------------------------------
# Exercise 3
# -----------------------------------------------------------------------------

# pi doesn't need to be imported in R
h = 5.0 # height 
b = 2.0 # base
r = 1.5 # radius
area_parallelogram = h*b
cat("The area of the parallelogram is" , area_parallelogram)
cat("\n")
area_square = b^2  # the power symbol in R is ^, in Python it's **
cat("The area of the square is" , area_square)
cat("\n")
area_circle = pi*r^2
cat("The area of the circle is" , area_circle)
cat("\n")
volume_cone = 1.0/3*pi*r^2*h
cat("The volume of the cone is" , volume_cone) #All of the print statements had to be translated from Python into R

# -----------------------------------------------------------------------------
# Exercise 4
# -----------------------------------------------------------------------------

# -------------------------------
# [4.1] 
# -------------------------------
x = pi/4
val_1 = sin(x)^2 + cos(x)^2 
cat("val_1= ", val_1) # The value turns out to be 1

# You can also try:
1 == sin(x)^2 + cos(x)^2
# The output of this will tell you whether the expression is true or false.
# Read about boolean type variables in your book if you want to know why this
# works

# -------------------------------
# [4.2] 
# -------------------------------
v0 = 3  # in m/s
t = 1   # in s
a = 2   # in m/s^2
s = v0*t + 1.0/2*(a*t^2) # In R you don't need to worry about 1/2 vs. 1.0/2.0, but
# it's good practice to do it anyway
cat("s= ", s)  

# -------------------------------
# [4.2] 
# -------------------------------
a = 3
b = 5
a2 = a^2 
b2 = b^2
eq1_sum = a2 + 2*a*b + b2 #added multiplication signs to 2ab
eq2_sum = a2 - 2*a*b + b2
eq1_pow = (a + b)^2
eq2_pow = (a - b)^2
cat("First equation:", eq2_sum %% eq2_sum) #Added extra % to find modulus
cat("\n")
cat("Second equation:", eq2_pow %% eq2_pow)
cat("\n")
# You can then do it again with
a = 3
b = 3

# A more elegant way is to use vector evaluation
a = c(3,3) 
b = c(5,3)
a2 = a^2
b2 = b^2
eq1_sum = a2 + 2*a*b + b2
eq2_sum = a2 - 2*a*b + b2
eq1_pow = (a + b)^2
eq2_pow = (a - b)^2
all.equal(eq1_sum, eq1_pow)
all.equal(eq2_sum, eq2_pow)
# or
eq1_sum == eq1_pow
eq2_sum == eq2_pow



# -----------------------------------------------------------------------------
# Exercise 5
# -----------------------------------------------------------------------------

# I printed the final results and checked
# whether it sums to zero. You can see that it does.
a=2
b=1
c=2   # This exercise has a complex number, which calls for the complex() command.
q=complex(real=0,imaginary=sqrt(15))   # The complex command in R, not it's
sqrt(15) not just 15!

x_1 = (-b + q)/(2.0*a)   # add parentheses for precedence
x_2 = (-b - q)/(2.0*a)   
cat(x_1)
cat("\n")
cat(x_2)
cat("\n")
cat("a*x_1^2 + b * x_1 + c=", a*x_1^2 + b * x_1 + c)
cat("\n")
cat("a*x_2^2 + b * x_2 + c=", a*x_2^2 + b * x_2 + c)

File: jdocs/Comp/Lectures/Lecture1/Homework/Python/Homework1.py
# -----------------------------------------------------------------------------
# Exercise 1
# -----------------------------------------------------------------------------

# If we want to use the array command, we need to import it first
from pylab import *

v1 = array([1,2,3,4,5,6,7,8,7,6,5,4,3,2,1]) # vector in array form
print "v1= " + str(v1)

v2 = array([1,2,2,3,3,3,4,4,4,4,5,5,5,5]) # vector in array form
print "v2= " + str(v2)

m1 = array([[0,1,1],[1,0,1],[1,1,0]]) # matrix in multi-dimensional array form
print "m1= " + str(m1)

m2 = array([[0,2,3],[0,5,0],[7,0,0]]) # matrix in multi-dimensional array form
print "m2= " + str(m2)



# -----------------------------------------------------------------------------
# Exercise 2
# -----------------------------------------------------------------------------
final_amount=0
p=5.0 #interest rate in percent
initial_amount=1000.00 #initial amount given in Euros
n=3
final_amount=initial_amount*(1.0+p/100.0)**n
print 'The final amount is', final_amount



# -----------------------------------------------------------------------------
# Exercise 3
# -----------------------------------------------------------------------------
from math import pi
h = 5.0 # height
b = 2.0 # base
r = 1.5 # radius
#
area_parallelogram = h*b
print "The area of the parallelogram is %.3f" % area_parallelogram
#
area_square = b**2
print "The area of the square is %g" % area_square
#
area_circle = pi*r**2
print "The area of the circle is %.3f" % area_circle
#
volume_cone = 1.0/3*pi*r**2*h
print 'The volume of the cone is %.3f' % volume_cone

# -----------------------------------------------------------------------------
# Exercise 4
# -----------------------------------------------------------------------------

# -------------------------------
# [4.1]
# -------------------------------
from math import pi, sin, cos
x = pi/4.0
val_1 = sin(x)**2 + cos(x)**2
print val_1

# -------------------------------
# [4.2]
# -------------------------------
v0 = 3.0
t = 1
a = 2
s = v0*t + 1.0/2.0 * a*t**2 # Note: float(1/2) does also produce 0.0 which is incorrect.

print "s= " +str(s)     # Don't forget to transform out s into string for printing

# -------------------------------
# [4.3]
# -------------------------------
a=3.0
b=5.0
a2 = a**2
b2 = b**2
eq1_sum = a2 + 2*a*b + b2  # Don't forget times: *
eq2_sum = a2 - 2*a*b + b2
eq1_pow = (a + b)**2
eq2_pow = (a - b)**2
print 'First equation:', eq2_sum % eq2_pow
print 'Second equation:', eq1_sum % eq1_pow

# And do it again for
a=3.0
b=3.0

# Or you use vector evaluations
a=array([3.0, 3.0])
b=array([5.0, 3.0])
a2 = a**2
b2 = b**2
eq1_sum = a2 + 2*a*b + b2  # had to put * in between 2, a, and b
eq2_sum = a2 - 2*a*b + b2
eq1_pow = (a + b)**2
eq2_pow = (a - b)**2

print "eq1_sum == eq1_pow: " + str(eq1_sum == eq1_pow)
print "eq2_sum == eq2_pow: " + str(eq2_sum == eq2_pow)

# -----------------------------------------------------------------------------
# Exercise 5
# -----------------------------------------------------------------------------

a=2; b=1; c=2
from math import sqrt
q=complex(0,sqrt(15))   # q=sqrt(b*b - 4*a*c) which becomes q=sqrt(-15) = 0 + sqrt(15)*i

x_1 = (-b + q)/(2*a)   # added parentheses for precedence
x_2 = (-b - q)/(2*a)
print "a*x_1^2 + b*x_1 + c= ", a*x_1**2 + b * x_1 + c
print "a*x_2^2 + b*x_2 + c= ", a*x_2**2 + b * x_2 + c
